distributed/utils_test.py

```diff
@@ -733,14 +731,13 @@ def cluster(
             for proc in [w["proc"] for w in workers]:
                 proc.join(timeout=30)
 
+            for worker in workers:
+                with suppress(OSError):
```

Comment:
this OSError means there's probably left over files, TemporaryDirectory uses rmtree's onerror to clean up 

https://github.com/python/cpython/blob/68e80c4e74fdf549146c2b724588038160d79d26/Lib/tempfile.py#L804-L832

Reply:
I switched to the ExitStack as you suggested